src/components/Select/Select.js

```diff
@@ -41,50 +40,95 @@ const baseSelectStyles = ({ theme }) => css`
   }
 `;
 
-const disabledSelectStyles = ({ disabled }) =>
-  disabled &&
+const selectStretchStyles = ({ stretch }) =>
+  stretch &&
   css`
-    label: select--disabled;
-    opacity: 0.4;
-    pointer-events: none;
+    label: select--stretch;
+    width: 100%;
   `;
 
-const baseIconStyles = ({ theme }) => css`
+const iconBaseStyles = ({ theme }) => css`
   label: select__icon;
   fill: ${theme.colors.n700};
   display: block;
   z-index: 40;
   pointer-events: none;
+  position: absolute;
+  ${size(theme.spacings.kilo)};
+  top: 50%;
+  right: ${theme.spacings.kilo};
+  transform: translateY(-50%);
+`;
+
+const containerBaseStyles = ({ theme }) => css`
+  label: select__container;
+  color: ${theme.colors.n900};
+  display: block;
+  position: relative;
+`;
+
+const containerDisabledStyles = ({ disabled }) =>
+  disabled &&
+  css`
+    label: select__container--disabled;
+    ${disableVisually()};
+  `;
+
+const containerInlineStyles = ({ theme, inline }) =>
+  inline &&
+  css`
+    label: select__container--inline;
+    display: inline-block;
+    margin-right: ${theme.spacings.mega};
+  `;
+
+const containerMarginStyles = ({ theme, margin }) =>
+  margin &&
+  css`
+    label: select__container--margin;
+    margin-bottom: ${theme.spacings.mega};
+  `;
+
+const SelectContainer = styled('div')`
+  ${containerBaseStyles};
+  ${containerMarginStyles};
+  ${containerDisabledStyles};
+  ${containerInlineStyles};
 `;
 
 const SelectElement = styled('select')`
-  ${baseSelectStyles} ${disabledSelectStyles};
+  ${selectBaseStyles};
+  ${selectStretchStyles};
 `;
 const Icon = styled(DownTriangleIcon)`
-  ${baseIconStyles};
+  ${iconBaseStyles};
 `;
 
 /**
  * A native select component.
  */
-const Select = ({ options, value, placeholder, disabled, ...props }) => (
-  <IconInputWrapper
-    iconPosition="right"
-    icon={({ className, disabledClassName }) => (
-      <Icon className={cx(className, { [disabledClassName]: disabled })} />
-    )}
-    input={({ className }) => (
-      <SelectElement {...{ ...props, className, disabled }}>
-        {!value && <option key={0}>{placeholder}</option>}
-        {options &&
-          options.map(option => (
-            <option key={option.value} {...option}>
-              {option.label}
-            </option>
-          ))}
-      </SelectElement>
-    )}
-  />
+const Select = ({
+  options,
+  value,
+  placeholder,
+  disabled,
+  margin,
+  inline,
+  stretch,
+  ...props
+}) => (
+  <SelectContainer {...{ margin, inline, disabled }}>
+    <SelectElement {...{ ...props, disabled, stretch }}>
```

Comment:
Hm, if I remember correctly, some feedback from @mlent was that these options shouldn't be a config, but rather children. I guess I did not address that. 😁 Would you mind?

Reply:
I was wondering about that. Will do :)